core/corehttp/gateway_handler.go

```diff
@@ -153,12 +154,24 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	nd, err := core.Resolve(ctx, i.node, path.Path(urlPath))
-	// If node is in offline mode the error code and message should be different
-	if err == core.ErrNoNamesys && !i.node.OnlineMode() {
-		w.WriteHeader(http.StatusServiceUnavailable)
-		fmt.Fprint(w, "Could not resolve path. Node is in offline mode.")
+	switch err {
+	case nil:
+		// core.Resolve worked
+	case namesys.ErrResolveFailed:
+		// Don't log that error as it is just noise
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
400 doesn't fit here since the 4xx range is for client errors, where the client can amend the request to possibly get a successful response. 502 Bad Gateway or, (less specific) 500 Internal Server Error would be sufficient.